ocpp/v201/datatypes.py

```diff
@@ -0,0 +1,809 @@
+    """
+
+    event_id: int
+    timestamp: datetime
+    trigger: enums.EventTriggerType
+    actual_value: str
+    event_notification_type: enums.EventNotificationType
+    component: ComponentType
+    variable: VariableType
+    cause: Optional[int] = None
+    tech_code: Optional[str] = None
+    cleared: Optional[bool] = None
+    transaction_id: Optional[str] = None
+    variable_monitoring_id: Optional[int] = None
+
+
+@dataclass
+class FirmwareType:
+    """
+    Represents a copy of the firmware that can be loaded/updated on the
+    Charging Station.
+    FirmwareType is used by: UpdateFirmwareRequest
+    """
+
+    location: str
+    retrival_date_time: datetime
+    install_date_time: Optional[datetime] = None
+    signing_certificate: Optional[str] = None
+    signature: Optional[str] = None
+
+
+@dataclass
+class GetVariableDataType:
+    """
+    Class to hold parameters for GetVariables request.
+    GetVariableDataType is used by: GetVariablesRequest
+    """
+
+    component: ComponentType
+    variable: VariableType
+    attribute_type: Optional[enums.AttributeType] = None
+
+
+@dataclass
+class GetVariableResultType:
+    """
+    Class to hold results of GetVariables request.
+    GetVariableResultType is used by: GetVariablesResponse
+    """
+
+    attribute_status: enums.GetVariableStatusType
+    component: ComponentType
+    attribute_type: Optional[enums.AttributeType] = None
+    attribute_value: Optional[str] = None
+
+
+@dataclass
+class IdTokenType:
+    """
+    Contains a case insensitive identifier to use for the authorization and the
+    type of authorization to support multiple forms of identifiers.
+    IdTokenType is used by: AuthorizationData, Common:IdTokenInfoType,
+    RequestStartTransactionRequest, AuthorizeRequest, TransactionEventRequest,
+    ReserveNowRequest, CustomerInformationRequest
+    """
+
+    id_token: str
+    type: enums.IdTokenType
+    additonal_info: Optional[AdditionalInfoType] = None
+
+
+@dataclass
+class MessageContentType:
+    """
+    Contains message details, for a message to be displayed on a Charging
+    Station.
+    MessageContentType is used by: IdTokenInfoType, MessageInfoType,
+    TransactionEventResponse
+    """
+
+    format: enums.MessageFormatType
+    content: str
+    language: Optional[str] = None
+
+
+@dataclass
+class IdTokenInfoType:
+    """
+    Contains status information about an identifier. It is advised to not stop
+    charging for a token that expires during charging, as ExpiryDate is only
+    used for caching purposes. If ExpiryDate is not given, the status has no
+    end date.
+    IdTokenInfoType is used by: AuthorizationData, AuthorizeResponse,
+    TransactionEventResponse
+    """
+
+    status: enums.AuthorizationStatusType
+    cache_expiry_date_time: Optional[datetime] = None
+    charging_priortiy: Optional[int] = None
+    language1: Optional[str] = None
+    evse_id: Optional[int] = None
+    language2: Optional[str] = None
```

Comment:
`language_1` and `language_2` would be the correct snake_casee spelling. However, I think the the [code] that handles conversion from snake_case to CamelCase and vice versa would do that correctly.  Can you verify that and update these methods if needed?

[code]: https://github.com/mobilityhouse/ocpp/blob/41d4a16c368cab124264a052844332d29248372e/ocpp/charge_point.py#L17-L69

Reply:
Can confirm that it gets converted correctly in snake_case.